DEVELOPMENT.md

```diff
@@ -197,6 +197,112 @@ kubectl delete pvc --all -A
 ./scripts/local-cluster.sh delete <name>
 ```
 
+## Modules
+
+Modules is the new way components are packaged and managed in Welkin.
+They are packaged as [Crossplane Configurations](https://docs.crossplane.io/latest/packages/configurations/) and installed as [Crossplane Composite Resources](https://docs.crossplane.io/latest/composition/composite-resources/).
+
+### Adding a Module
+
+These are the general steps to do when replacing old Helm chart(s) with a Module.
+
+1. Add the Crossplane Configuration package to `helmfile.d/lists/images.yaml`.
+
+    ```diff
+    images:
+      crossplane:
+        configurations:
+    +      module-$NAME: $REGISTRY/welkin/module-$NAME:$VERSION
+    ```
+
+1. Create a Module Helm chart in `helmfile.d/charts/modules`.
+
+    ```sh
+    mkdir helmfile.d/charts/modules/$NAME
+    ```
+
+1. Add the `Chart.yaml` file in `helmfile.d/charts/modules/$NAME`.
+
+    ```yaml
+    apiVersion: v2
+    name: module-$NAME
+    description: A Helm chart for the $NAME Module
+    type: application
+    version: 0.1.0
+    appVersion: $APPLICATION_VERSION
+    ```
+
+1. Copy the values file from the old chart to the Module chart.
+
+> [!note]
+> The values does not need to be identical to the previous chart but everything that was previously configurable must still be configurable.
```

Comment:
I feel like this would be a better wording as the old chart isn't really the point here is it, it is rather what can be set via the config?
```suggestion
> The values does not need to be identical to the previous chart but everything that is already configurable via the Apps common/sc/wc-config must still be configurable through the module.
```

Reply:
Great suggestion, thanks!